packages/eslint-plugin/src/rules/prefer-return-this-type.ts

```diff
@@ -147,12 +147,14 @@ export default createRule({
       }
 
       if (isFunctionReturningThis(originalFunc, originalClass)) {
+        const node =
+          classNameRef.parent?.type === AST_NODE_TYPES.TSTypeReference
+            ? classNameRef.parent
+            : classNameRef;
```

Comment:
instead of doing this here - we should instead change the `tryGetNameInType` function to return the `TSTypeReference` instead of the identifier.